pkg/controller/apiserver/apiserver_controller.go

```diff
@@ -72,16 +71,10 @@ func Add(mgr manager.Manager, opts options.AddOptions) error {
 
 	// Established deferred watches against the v3 API that should succeed after the Enterprise API Server becomes available.
 	if opts.EnterpriseCRDExists {
-		k8sClient, err := kubernetes.NewForConfig(mgr.GetConfig())
```

Comment:
This code to create a new k8s clientset is repeated across almost all controllers. It has now been refactored to be initialized once in the main function and passed as port of opts.